lib/file_store/adapters/s3.ex

```diff
@@ -151,9 +151,13 @@ if Code.ensure_loaded?(ExAws.S3) do
     defp get_region(store), do: store |> get_config() |> Map.fetch!(:region)
     defp get_config(store), do: ExAws.Config.new(:s3, get_overrides(store))
     defp get_overrides(store), do: Map.get(store.config, :ex_aws, [])
-    defp get_prefix(store), do: Map.get(store.config, :prefix, "")
 
-    defp prefix_key(store, key), do: Path.join(get_prefix(store), key)
+    defp prefix_key(store, key) do
```

Comment:
The reason that I used `Path.join` is that it _works_ like a path:

```elixir
Path.join("", "foo") # => "foo"
Path.join("foo", "bar") # => "foo/bar"
```

I know that S3 isn’t really a path-based system, but given that prefixes are always separated with `/`, it’s really indistinguishable.

Reply:
Yeah, it was a totally reasonable way to solve the problem. I had it in my head that `Path.join` would use a `\` on Windows, but after looking into it a bit more, I think my assumption may have been incorrect.

Thanks again for all of these PRs. They're incredibly valuable and I'm so sorry it took me so long to get around to merging them.

Reply:
No problem. Until I make the modification to my use of S3 to convert to binary in advance, I can’t move from my `omnibus` custom branch, but this is a _really_ nice set of abstractions and it was basically something that I was going to end up writing. Funnily enough, I ended up a second set of abstractions on top of it because in my use case, I was dealing with a database record that wrapped the stored filenames.